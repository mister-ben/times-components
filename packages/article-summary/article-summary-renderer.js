import React from "react";
import { Text } from "react-native";

export default {
  paragraph(key, attributes, renderedChildren) {
    return <Text key={key}>{renderedChildren} </Text>;
  },
  sentence(key, attributes, renderedChildren) {
    return <Text key={key}>{renderedChildren}. </Text>;
  },
  teaser(key, attributes, renderedChildren) {
    return <Text key={key}>{renderedChildren}...</Text>;
  }
};
