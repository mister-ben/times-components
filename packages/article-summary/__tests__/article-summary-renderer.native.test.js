/* eslint-env jest */
import { Text, View } from "react-native";
import runTests from "./test-helper";
import { renderer } from "../article-summary-renderer";

jest.mock("WebView", () => "WebView");
describe("Article Summary Renderer Native", runTests(renderer));
