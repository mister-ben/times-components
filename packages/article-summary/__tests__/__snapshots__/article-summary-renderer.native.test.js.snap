// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Summary Renderer Native renders a paragraph ending with a full stop and a space with no new line 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Content
  . 
</Text>
`;

exports[`Article Summary Renderer Native renders a teaser with an ellipsis at the end 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Content
  ...
</Text>
`;
